src/detect-engine-analyzer.c

```diff
@@ -861,6 +863,22 @@ static void DumpMatches(RuleAnalyzer *ctx, JsonBuilder *js, const SigMatchData *
                 jb_close(js);
                 break;
             }
+            case DETECT_TCPMSS: {
+                const DetectU16Data *cd = (const DetectU16Data *)smd->ctx;
+
+                jb_open_object(js, "tcp_mss");
+                const char *flag = DetectTcpmssModeToString(cd->mode);
+                jb_set_string(js, "operand", flag);
+                if (strcmp(flag, "range") == 0) {
```

Comment:
I would use the `cd->mode` integer for comparison instead of its derivated string ;-)

Reply:
It's what I used at first; I was asked to change it.

So!

Reply:
Where ? I did not find it in previous PRs...

Reply:
Wait! Are you talking about the “range” operand?

Reply:
Oh! Since `mode` had been converted to string, doing the comparison with `strcmp` was the only way.

Reply:
you still have access to `cd->mode` ;-) (and it is cheaper to compare integer than strings)

Reply:
True True.